src/Examples/UnitModels/cstr_testing.ipynb

```diff
@@ -0,0 +1,1123 @@
+    "m.fs.heating_cost = Expression(expr=2.2e-7 * m.fs.H101.heat_duty[0])  # the stream must be heated to T_rxn, so H101 duty is positive\n",
+    "m.fs.operating_cost = Expression(expr=(3600 * 8000 *(m.fs.heating_cost + m.fs.cooling_cost)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fixing feed conditions\n",
+    "\n",
+    "Let us first check how many degrees of freedom exist for this flowsheet using the `degrees_of_freedom` tool we imported earlier. We expect each stream to have 6 degrees of freedom, the mixer to have 0 (after both streams are accounted for), the heater to have 1 (just the duty, since the inlet is also the outlet of M101), and the reactor to have 1 (duty or conversion, since the inlet is also the outlet of H101). In this case, the reactor has an extra degree of freedom (reactor conversion or reactor volume) since we have not yet defined the CSTR performance equation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "15\n",
+      "0\n",
+      "1\n",
+      "2\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(degrees_of_freedom(m))  # (6 per stream x 2 streams) + (3 from units) = 15 DOF\n",
+    "print(degrees_of_freedom(m.fs.M101) - 12)  # 2 inlets, so subtract those 12 DOF\n",
+    "print(degrees_of_freedom(m.fs.H101) - 6)   # 1 inlet, so subtract those 6 DOF\n",
+    "print(degrees_of_freedom(m.fs.R101) - 6)   # 1 inlet, so subtract those 6 DOF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "tags": [
+     "testing",
+     "remove_cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Check the degrees of freedom\n",
+    "assert degrees_of_freedom(m) == 15\n",
+    "assert degrees_of_freedom(m.fs.M101) - 12 == 0\n",
+    "assert degrees_of_freedom(m.fs.H101) - 6 == 1\n",
+    "assert degrees_of_freedom(m.fs.R101) - 6 == 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now be fixing the feed stream to the conditions shown in the flowsheet above. As mentioned in other tutorials, the IDAES framework expects a time index value for every referenced internal stream or unit variable, even in steady-state systems with a single time point $ t = 0 $. The non-present components in each stream are assigned a very small non-zero value to help with convergence and initializing. Based on stoichiometric ratios for the reaction, 80% conversion and 200 MM lb/year (46.4 mol/s) of ethylene glycol, we will initialize our simulation with the following calculated values:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.fs.M101.reagent_feed.flow_mol_phase_comp[0, \"Liq\", \"ethylene_oxide\"].fix(58.0)\n",
```

Comment:
I believe we can no do the following:

```
m.fs.M101.reagent_feed.flow_mol_phase_comp[0, "Liq", "ethylene_oxide"].fix(58.0*pyunits.mol/pyunits.s)
```

I.e. we can include units when fixing/setting variable values. If this works, we should begin doing it throughout our examples to properly show units and the formal way to do things.

Reply:
This syntax does work, I will update all of the specifications to explicitly include units and do this going forward as well.